web/pandas/pdeps/0013-logical-type-system.md

```diff
@@ -0,0 +1,197 @@
+
+The methods of each logical type are expected in turn to yield another logical type. This can enable us to smooth over differences between the NumPy and Arrow world, while also leveraging the best of both backends. To illustrate, let's look at some methods where the return type today deviates for end users depending on if they are using NumPy-backed data types or Arrow-backed data types. The equivalent PDEP-13 logical data type is presented as the last column:
+
+| Method             | NumPy-backed result | Arrow Backed result type | PDEP-13 result type            |
+|--------------------|---------------------|--------------------------|--------------------------------|
+| Series.str.len()   | np.float64          | pa.int64()               | pd.Int64Dtype()                |
+| Series.str.split() | object              | pa.list(pa.string())     | pd.ListDtype(pa.StringDtype()) |
+| Series.dt.date     | object              | pa.date32()              | pd.DateDtype()                 |
+
+The ``Series.dt.date`` example is worth an extra look - with a PDEP-13 logical type system in place we would theoretically have the ability to keep our default ``pd.DatetimeDtype()`` backed by our current NumPy-based array but leverage pyarrow for the ``Series.dt.date`` solution, rather than having to implement a DateArray ourselves.
+
+While this PDEP proposes reusing existing extension types, it also necessitates extending those types with extra metadata:
+
+```python
+class BaseType:
+
+    @property
+    def data_manager -> Literal["numpy", "pyarrow"]:
+        """
+        Who manages the data buffer - NumPy or pyarrow
+        """
+        ...
+
+    @property
+    def physical_type:
+        """
+        For logical types which may have different implementations, what is the
+        actual implementation? For pyarrow strings this may mean pa.string() versus
+        pa.large_string() versrus pa.string_view(); for NumPy this may mean object
+        or their 2.0 string implementation.
+        """
+        ...
+
+    @property
+    def na_marker -> pd.NA|np.nan|pd.NaT:
+        """
+        Sentinel used to denote missing values
+        """
+        ...
+```
+
+``na_marker`` is expected to be read-only (see next section). For advanced users that have a particular need for a storage type, they may be able to construct the data type via ``pd.StringDtype(data_manager=np)`` to assert NumPy managed storage. While the PDEP allows constructing in this fashion, operations against that data make no guarantees that they will respect the storage backend and are free to convert to whichever storage the internals of pandas considers optimal (Arrow will typically be preferred).
+
+### Missing Value Handling
```

Comment:
@jbrockmendel wanted to ping this off you. When we talked I said would stick with np.nan as the missing value marker, but since we are reusing the `pd.StringDtype()` in this version of the proposal I don't think we should move that away from `pd.NA` 

Its not the default but maybe this ``pd.na_marker="legacy"`` serves as a good middle ground?

Reply:
if there's a global option, it should go through pd.set_option/get_option

Reply:
why data_manager instead of storage?

Reply:
The only reason I wanted to include the word data is because we are strictly referring to the data buffer and nothing else. I don't have a strong preference on the terminology of this though